src/pipeline.ts

```diff
@@ -816,6 +938,52 @@ function snakeCaseKeys<T = unknown>(obj: T, sep = '-'): T {
   return result as any;
 }
 
+/**
+ * Multiple stages that are deployed in parallel
+ *
+ * A `Wave`, but with addition GitHub options
+ *
+ * Create with `GitHubWorkflow.addWave()` or `GitHubWorkflow.addGitHubWave()`, do not construct directly
+ */
+export class GitHubWave extends Wave {
+  constructor(
+    /** Identifier for this Wave */
+    public readonly id: string,
+    /** GitHubWorkflow that this wave is part of  */
+    private pipeline: GitHubWorkflow,
+    props: WaveProps = {},
+  ) {
+    super(id, props);
+  }
+
+  /**
+   * Add a Stage to this wave
+   *
+   * It will be deployed in parallel with all other stages in this
+   * wave.
+   */
+  public addStage(stage: Stage, options: AddStageOpts = {}) {
```

Comment:
Is there anyway where we can have `addStage` call `addStageWithGitHubOptions` or the other way around. I don't like having two very similar methods co-existing. If we ever want to change one of the methods, we'd have to remember to change the other

Reply:
Makes sense. I'll have `addStage` call `this.addStageWithGitHubOptions(stage, options)` - no code duplication there.